### PR TITLE
Fyst 807 1099 r create base table and polymorphic tables

### PR DESCRIPTION
**Pull request by mrotondo**
## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-807
## Is PM acceptance required? (delete one)
- No - merge after code review approval
## What was done?
- Added a StateFile1099R table, and gave it polymorphic references to the existing StateFileXxIntake tables and to new StateFileXx1099RFollowup tables. 
- Implemented synchronizing of 1099Rs in the DF XML over to the table when importing from an API response or submitting the federal info form
- Removed the previously-added 1099R aggregate fields from StateFileAzIntake along with the associated controller/form/view
## How to test?
- Start an AZ intake, select an XML with at least one 1099R, select "i want to edit XML" & note that it's on the federal info form

### CODE REVIEW

File: spec/fixtures/state_file/fed_return_xmls/2023/az/richard_retirement_1099r.xml [hunk 1]
```diff
@@ -114,7 +114,7 @@
         <F1099RStateTaxGrp>
           <StateTaxWithheldAmt>11</StateTaxWithheldAmt>
           <StateAbbreviationCd>NC</StateAbbreviationCd>
-          <PayerStateIdNumber>111111111</PayerStateIdNumber>
+          <PayerStateIdNum>111111111</PayerStateIdNum>
```
Comment (mpidcock) [inline spec/fixtures/state_file/fed_return_xmls/2023/az/richard_retirement_1099r.xml:117]: I see in the federal schema that it uses `PayerStateIdNum`... but the xml with `PayerStateIdNumber` was provided to us from the DF team. This change makes sense to me, I'm just confused about how it happened.

File: app/models/state_file1099_r.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#  payer_state_identification_number  :string
+#  payer_zip                          :string
+#  phone_number                       :string
+#  recipient_name                     :string
+#  recipient_ssn                      :string
+#  standard                           :boolean
+#  state_code                         :string
+#  state_distribution_amount          :integer
+#  state_specific_followup_type       :string
+#  state_tax_withheld_amount          :integer
```
Comment (mpidcock) [inline app/models/state_file1099_r.rb:28]: With our new monetary input fields, at a minimum `state_distribution_amount` and `state_tax_withheld_amount` will need to be decimals. 

A case could be made that the other monetary amounts also should be decimals, but if they aren't being edited it's not a strict requirement. (though I would lean towards all decimals anyway)

**Review by mpidcock — COMMENTED** [review]
It all looks great except for needing decimal fields for amounts

File: spec/fixtures/state_file/fed_return_xmls/2023/az/richard_retirement_1099r.xml (hunk 1 above)
Comment (mrotondo) [reply spec/fixtures/state_file/fed_return_xmls/2023/az/richard_retirement_1099r.xml:117]: Resolved by looking w/ martha at the most recent sample XML from kiteworks & observing that it uses the element name from the federal XML schema (PayerStateIdNum)

**Review by mpidcock — APPROVED** [review]
🔢

**Pull request merged**